lib/graphql/batch/setup.rb

```diff
@@ -1,14 +1,54 @@
 module GraphQL::Batch
-  module Setup
-    extend self
+  class Setup
+    class << self
+      def start_batching
+        raise NestedError if GraphQL::Batch::Executor.current
+        GraphQL::Batch::Executor.current = GraphQL::Batch::Executor.new
+      end
+
+      def end_batching
+        GraphQL::Batch::Executor.current = nil
+      end
+
+      def instrument_field(schema, type, field)
+        return field unless type == schema.mutation
+        old_resolve_proc = field.resolve_proc
+        field.redefine do
+          resolve ->(obj, args, ctx) {
+            GraphQL::Batch::Executor.current.clear
```

Comment:
Not totally sure why you need to clear before and after?

Reply:
Clearing afterwards is to avoid using data cached from before the mutation.  Clearing before the mutation isn't as important, but just ensures that the field is actually executed serially.  However, I could see how it might be desirable to use the already loaded values in the mutation.

Do you think we should clear before the mutation?  Or get rid of this one?